app/queries/create_post_mutator.rb

```diff
@@ -36,6 +36,8 @@ def create_post
       # Send a notification mail to addressee only if she isn't replying to herself.
       UserMailer.new_post(post, addressee).deliver_later if addressee.present? && current_user != addressee
 
+      Notifications::PostCreatedJob.perform_later(current_user.id, post.id)
```

Comment:
@bodhish Have you considered using some kind of pub-sub mechanism here instead of directly scheduling PostCreatedJob? 

The context is:
We are planning to start a project to add integration of PF LMS with LRS (Learning Record Store) via xAPI.
The solution is not ready yet but our concept is to use some pub-sub mechanism to push events when some interesting things happen in PF domain (aka domain events) and react to them in some subscriber to process the fact, build xAPI statement and push it to the LRS.

It looks we could use the same pub-sub mechanism here and build notifications also in a handler (build as a subscriber of pub-sub mechanism).

I would not hardcode any pub-sub mechanism into PF code - I would prefer to have it injected into mutators and other parts of system to allow switching it in forked projects if necessary.

If you would like to discuss this please post the comment here or ping me on discord.

Reply:
Will ping you on discord.

Reply:
@mpraglowski I had a conversation with @bodhish about this and we're of the opinion that instead of introducing a _new_ pub-sub mechanism, we should consider extending the abilities of the webhook feature that we have on PF.

Right now, the webhook system is tied to individual courses, and it only allows one webhook per course. Instead, we believe it would make sense to upgrade webhooks so that they're linked to the`School` (the top-level entity), and to allow any number of webhooks, so that multiple webhooks can be configured to handle different events.

With that in place, it should be possible to decouple responsibilities by emitting a new `post.created` event from this location. The webhook receiver could then be implemented anywhere, including as new routes in forked projects.

If you have think this approach has any disadvantages, please do ping us on Discord. We can discuss on the `#general` channel!

Reply:
Ok. let's move to discord :)